examples/layout.rs

```diff
@@ -0,0 +1,143 @@
+                    draggable: true,
+                    initial_position: Vec2::new(10.0, 10.0),
+                    size: Vec2::new(512.0, 512.0),
+                    ..KWindow::default()
+                }}
+            >
+                <ElementBundle
+                    styles={KStyle{
+                        layout_type: LayoutType::Grid.into(),
+                        grid_rows: vec![Units::Stretch(1.0), Units::Stretch(2.0), Units::Stretch(5.0)].into(),
+                        grid_cols: vec![Units::Stretch(1.0), Units::Stretch(1.0)].into(),
+                        ..default()
+                    }}
+                >
+                    <BackgroundBundle
+                        styles={KStyle{
+                            background_color: Color::rgb(0.4, 0.9, 0.4).into(),
+                            color: Color::rgb(0.0, 0.0, 0.0).into(),
+                            padding: Edge::all(Units::Pixels(5.0)).into(),
+                            border_radius: Corner::all(10.0).into(),
+                            row_index: 0.into(),
+                            col_index: 0.into(),
+                            col_span: 2.into(),
+                            layout_type: LayoutType::Row.into(),
+                            col_between: Units::Pixels(5.0).into(),
+                            ..default()
+                        }}
+                    >
+                        <TextWidgetBundle
+                            text={TextProps {
+                                content: "A".into(),
+                                ..default()
+                            }}
+                        />
+                        <TextWidgetBundle
+                            text={TextProps {
+                                content: "B".into(),
+                                ..default()
+                            }}
+                        />
+                        <TextWidgetBundle
+                            text={TextProps {
+                                content: "C".into(),
+                                ..default()
+                            }}
+                        />
+                        <TextWidgetBundle
+                            text={TextProps {
+                                content: "D".into(),
+                                ..default()
+                            }}
+                        />
+                        <TextWidgetBundle
+                            text={TextProps {
+                                content: "E".into(),
+                                ..default()
+                            }}
+                        />
+
+                    </BackgroundBundle>
+                    <TextWidgetBundle
+                        text={TextProps {
+                            content: "R1 C0".into(),
+                            ..default()
+                        }}
+                        styles={KStyle{
+                            row_index: 1.into(),
```

Comment:
This is a bit confusing and I'm hoping I can fix this in the future but text has some special requirements for styles which causes it to overwrite styles internally. The recommended way of setting styles on text is to use `user_styles` which is a field in `TextProps` I tested locally and this should solve the issue you saw. :)

Reply:
I had the same issue with `KButton` though. Do they also have this requirement?

Reply:
> I had the same issue with `KButton` though. Do they also have this requirement?

Not all of them but any widgets that mutate their own styles. I think what is going to happen is that widgets will be required to have a `computed_styles: KStyle` on all widgets that have styles. So that would look something like:
```rust
rsx! {
  <KButtonBundle
    styles={KStyle {
      width: Units::Pixels(240.0).into(),
      ..Default::default(),
    }}
  />
}

// In KButtonBundle
pub struct KButtonBundle {
  pub button: KButton,
  pub styles: KStyle,
  pub computed_styles: ComputedStyles,
  pub on_event: OnEvent,
  pub widget_name: WidgetName,
}

*computed_styles = KStyle::default()
  .with_style(KStyle {
      render_command: StyleProp::Value(RenderCommand::Quad),
      ..Default::default()
  })
  .with_style(styles.clone())
  .with_style(KStyle {
      background_color: Color::rgba(0.254, 0.270, 0.349, 1.0).into(),
      border_color: if state.hovering {
          hover_color.into()
      } else {
          Color::rgba(0.254, 0.270, 0.349, 1.0).into()
      },
      border: Edge::all(2.0).into(),
      border_radius: StyleProp::Value(Corner::all(10.0)),
      height: StyleProp::Value(Units::Pixels(28.0)),
      width: Units::Stretch(1.0).into(),
      cursor: StyleProp::Value(KCursorIcon(CursorIcon::Hand)),
      ..Default::default()
  }).into();
```

Reply:
@maccesch 
Please see #179 for a fix.

Reply:
nice work! It works now! Just pushed the updated branch.